pkg/products/threescale/reconciler.go

```diff
@@ -342,6 +392,349 @@ func (r *Reconciler) backupSystemSecrets(ctx context.Context, serverClient k8scl
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) getDeploymentConfig(ctx context.Context, client k8sclient.Client, dcName string) (*appsv1.DeploymentConfig, integreatlyv1alpha1.StatusPhase, error) {
+	apiCastDeploymentConfig := &appsv1.DeploymentConfig{}
+
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: dcName, Namespace: r.Config.GetNamespace()}, apiCastDeploymentConfig)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			return nil, integreatlyv1alpha1.PhaseAwaitingComponents, nil
+		}
+		return nil, integreatlyv1alpha1.PhaseFailed, fmt.Errorf("error getting the deployment config %v", err)
+	}
+	return apiCastDeploymentConfig, integreatlyv1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) getService(ctx context.Context, client k8sclient.Client, dcName string) (*corev1.Service, integreatlyv1alpha1.StatusPhase, error) {
+	service := &corev1.Service{}
+
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: dcName, Namespace: r.Config.GetNamespace()}, service)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			return nil, integreatlyv1alpha1.PhaseAwaitingComponents, nil
+		}
+		return nil, integreatlyv1alpha1.PhaseFailed, fmt.Errorf("error getting the deployment config %v", err)
+	}
+	return service, integreatlyv1alpha1.PhaseInProgress, nil
+
+}
+
+func (r *Reconciler) patchDeploymentConfig(ctx context.Context, client k8sclient.Client, deploymentConfig *appsv1.DeploymentConfig, service *corev1.Service) error {
+
+	ports := service.Spec.Ports
+	for i, port := range ports {
+		if port.Name == "gateway" {
+			service.Spec.Ports[i].Port = 8443
+			service.Spec.Ports[i].TargetPort = intstr.FromInt(8443)
+			break
+		}
+	}
+	if err := client.Update(ctx, service); err != nil {
+		return fmt.Errorf("failed to update service: %v", err)
+	}
+
+	deploymentConfig.Spec.Template.Labels["marin3r.3scale.net/status"] = "enabled"
+	deploymentConfig.Spec.Template.Annotations["marin3r.3scale.net/node-id"] = apicastRatelimiting
+	deploymentConfig.Spec.Template.Annotations["marin3r.3scale.net/ports"] = "envoy-https:8443"
+
+	if err := client.Update(ctx, deploymentConfig); err != nil {
+		return fmt.Errorf("failed to update deployment config: %v", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) createEnvoyRateLimitingConfig(ctx context.Context, client k8sclient.Client) error {
+	var podIp string
+
+	pods := &corev1.PodList{}
+
+	listOpts := []k8sclient.ListOption{
+		k8sclient.InNamespace("redhat-rhmi-marin3r"),
+		k8sclient.MatchingLabels(map[string]string{"app": "ratelimit"}),
+	}
+
+	err := client.List(ctx, pods, listOpts...)
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase == "Running" {
+			podIp = pod.Status.PodIP
+			break
+		}
+	}
```

Comment:
I'm probably missing something but asking just in case: Wouldn't it be better to point to the Service that's serving the ratelimit deployment? [1] This would also solve the issue with replication as the service could load balance the rate limit pods

1. https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/marin3r/rateLimitService.go#L259

Reply:
Thanks @sergioifg94, that idea works well!